addons/l10n_it_edi/models/res_config_settings.py

```diff
@@ -1,93 +1,55 @@
-
-    @api.depends('company_id.account_edi_proxy_client_ids', 'company_id.account_edi_proxy_client_ids.active')
-    def _compute_l10n_it_edi_proxy_current_state(self):
+    @api.depends('company_id')
+    def _compute_l10n_it_edi_show_purchase_journal_id(self):
         for config in self:
-            proxy_user = config.company_id.account_edi_proxy_client_ids.search([
-                ('company_id', '=', config.company_id.id),
-                ('proxy_type', '=', 'l10n_it_edi'),
-            ], limit=1)
-
-            config.l10n_it_edi_proxy_current_state = 'inactive' if not proxy_user else 'demo' if proxy_user.id_client[:4] == 'demo' else 'active'
+            # Only show the setting when there exists more than 1 purchase journal.
+            total_purchase_journal = self.env['account.journal'].search_count([
+                *self.env['account.journal']._check_company_domain(config.company_id),
+                ('type', '=', 'purchase'),
+            ])
+            config.l10n_it_edi_show_purchase_journal_id = total_purchase_journal >= 2
 
     @api.depends('company_id')
     def _compute_l10n_it_edi_register(self):
-        """Needed because it expects a compute"""
-        self.l10n_it_edi_register = False
-
-    def _set_l10n_it_edi_register_demo_mode(self):
         for config in self:
+            config.l10n_it_edi_register = config.company_id.l10n_it_edi_register
 
+    def _set_l10n_it_edi_register(self):
+        for config in self:
+            config.company_id.l10n_it_edi_register = config.l10n_it_edi_register
             proxy_user = self.env['account_edi_proxy_client.user'].search([
                 ('company_id', '=', config.company_id.id),
                 ('proxy_type', '=', 'l10n_it_edi'),
+                ('edi_mode', '!=', 'demo'),  # make sure it's a "real" proxy_user (edi_mode is 'test' or 'prod')
             ], limit=1)
 
-            real_proxy_users = self.env['account_edi_proxy_client.user'].sudo().search([
-                ('company_id', '=', config.company_id.id),
-                ('proxy_type', '=', 'l10n_it_edi'),
-                ('id_client', 'not like', 'demo'),
-            ])
-
-            # Update the config as per the selected radio button
-            previous_demo_state = proxy_user.edi_mode
-            edi_mode = config.l10n_it_edi_demo_mode
-
-            # If the user is trying to change from a state in which they have a registered official or testing proxy client
-            # to another state, we should stop them
-            if real_proxy_users and previous_demo_state != edi_mode:
-                raise UserError(_("The company has already registered with the service as 'Test' or 'Official', it cannot change."))
-
-            if config.l10n_it_edi_register:
-                # There should only be one user at a time, if there are no users, register one
+            if config.l10n_it_edi_register:  # Create a new proxy user or reactivate an existing one
```

Comment:
```suggestion
            if proxy_user:
                proxy_user._toggle_proxy_user_active()
            elif config.l10n_it_edi_register:
                edi_mode = self.env['ir.config_parameter'].sudo().get_param('l10n_it_edi.proxy_user_edi_mode') or 'prod'
                self._create_proxy_user(config.company_id, edi_mode)
```
Could we simplify the code like this?

Reply:
We can. But since `l10n_it_edi_register` may be set to the same value (e.g. False to False), I'll modify the check to be like this:

```py
            if proxy_user and proxy_user.active != config.l10n_it_edi_register:
                # deactivate / reactive the current proxy user based on the config's l10n_it_edi_register value
                proxy_user._toggle_proxy_user_active()
            elif config.l10n_it_edi_register and not proxy_user:
                # create a new proxy user
                edi_mode = self.env['ir.config_parameter'].sudo().get_param('l10n_it_edi.proxy_user_edi_mode') or 'prod'
                self._create_proxy_user(config.company_id, edi_mode)
```

Reply:
Looks good :+1: